sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -126,6 +126,8 @@ object JdbcDialects {
   registerDialect(MySQLDialect)
   registerDialect(PostgresDialect)
   registerDialect(DB2Dialect)
+  registerDialect(SqlServerDialect)
```

Comment:
Happy to make this and the other changes, I wasn't sure if MsSQLServer or SqlServer was better, I went with SqlServer as the JDBC url is "sqlserver".

I'll be away from the office for the next week but I'll submit an updated change when I'm back.

Thanks,
Ewan